benches/compare.rs

```diff
@@ -26,25 +28,29 @@ fn mutating_inputs(c: &mut Criterion) {
         codspeed_criterion_compat::measurement::WallTime,
     > = c.benchmark_group("Mutating Inputs");
 
-    let mut db = salsa::DatabaseImpl::default();
-
     for n in &[10, 20, 30] {
-        let base_string = "hello, world!".to_owned();
-        let base_len = base_string.len();
-
-        let string = base_string.clone().repeat(*n);
-        let new_len = string.len();
-
         group.bench_function(BenchmarkId::new("mutating", n), |b| {
-            b.iter(|| {
-                let input = Input::new(&db, base_string.clone());
-                let actual_len = length(&db, input);
-                assert_eq!(base_len, actual_len);
-
-                input.set_text(&mut db).to(string.clone());
-                let actual_len = length(&db, input);
-                assert_eq!(new_len, actual_len);
-            })
+            b.iter_batched_ref(
+                || {
+                    let db = salsa::DatabaseImpl::default();
+                    let base_string = "hello, world!".to_owned();
+                    let base_len = base_string.len();
+
+                    let string = base_string.clone().repeat(*n);
+                    let new_len = string.len();
+                    (db, base_string, base_len, string, new_len)
+                },
+                |(db, base_string, base_len, string, new_len)| {
+                    let input = Input::new(db, base_string.clone());
+                    let actual_len = length(db, input);
+                    assert_eq!(base_len, actual_len);
+
+                    input.set_text(db).to(string.clone());
+                    let actual_len = length(db, input);
+                    assert_eq!(new_len, actual_len);
```

Comment:
This seems incorrect now. I think the idea of the benchmark is to mutate `input` by changing the text and measuring how long it takes to do the incremental computation. 

What this is measuring now is how long it takes to create a new input and changing that input.

Reply:
Well, that was the behavior before my change already. You are right that the name implies we want to only bench the mutation but it never did so. Will change that

Reply:
[`99568ba` (#643)](https://github.com/salsa-rs/salsa/pull/643/commits/99568baff84b5b269accdce9b877a5a8644af485)